source/includes/fundamentals/code-snippets/bson-extra-elements/BsonExtraElementsExample.java

```diff
@@ -0,0 +1,74 @@
+package fundamentals;
+
+import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.ClassModel;
+import org.bson.codecs.pojo.Conventions;
+import org.bson.codecs.pojo.PojoCodecProvider;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.result.UpdateResult;
+
+public class BsonExtraElementsExample {
+
+	private static String DOC_NAME_VALUE = "MDB0123";
+
+    private static CodecRegistry getCodecRegistry() {
+        ClassModel<ProductWithBsonExtraElements> classModel = ClassModel.builder(ProductWithBsonExtraElements.class).
+                conventions(Arrays.asList(Conventions.ANNOTATION_CONVENTION)).build();
+        PojoCodecProvider pojoCodecProvider = PojoCodecProvider.builder().register(classModel).build();
+        return fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+    }
+
+    private static <T> void setup(MongoCollection<T> collection) {
```

Comment:
Don't make this generic.  Instead just take an instance of `MongoCollection<ProductWithBsonExtraElements>`

Reply:
I did this intentionally so other writers could use this class as a model for creating other small applications to test their code examples. Are there any disadvantages in using generic types for this use case?

Reply:
Yes, because the cast to `T` is unsafe and generates a warning.